subsys/tracing/include/tracing_user.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020 Lexmark International, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _TRACE_USER_H
+#define _TRACE_USER_H
+#include <kernel.h>
+#include <kernel_structs.h>
+#include <init.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void user_sys_trace_thread_switched_in(struct k_thread *thread);
```

Comment:
The namespace of these functions look a bit weird, should be place the `user` at the end of the func name, so
`sys_trace_thread_switched_in_user()`?

Reply:
it follows the same naming convention as ctf, so there was precedent for this.

Reply:
I agree that user_sys_ is weird, where is this the case for ctf?

Reply:
@nashif 
in ctf_top.c: sys_trace_thread_switched_in() calls ctf_top_thread_switched_in() (where the prefix identifies the subsystem)

I was trying to copy that by having sys_trace_thread_switched_in() call user_sys_trace_thread_switched_in().

Would user_thread_switched_in() be an acceptble name to maintain the prefix precedence?

Reply:
Moved the "user" to the end of the function names.  That seems to be a little more pleasing to the eyes.